conda/gateways/disk/__init__.py

```diff
@@ -63,3 +64,18 @@ def mkdir_p(path):
             return path
         else:
             raise
+
+
+def mkdir_p_sudo_safe(path):
+    if isdir(path):
+        return
+    base_dir = dirname(path)
+    if not isdir(base_dir):
+        mkdir_p_sudo_safe(base_dir)
+    log.trace('making directory %s', path)
+    mkdir(path)
```

Comment:
What if `mkdir()` throws an exception? Partition ran out of inodes or something ...